statistics/handle/lru_cache.go

```diff
@@ -0,0 +1,334 @@
+	}
+	l.calculateCost(tblMemUsage, &statistics.TableMemoryUsage{})
+	element = l.maintainList(nil, newCacheEntry, tblMemUsage, &statistics.TableMemoryUsage{})
+	l.elements[key] = element
+	return true
+}
+
+// Del deletes the key-value pair from the LRU Cache.
+func (l *internalLRUCache) Del(key int64) {
+	var del bool
+	var trackingCost int64
+	var totalCost int64
+	l.Lock()
+	del = l.del(key)
+	trackingCost = l.trackingCost
+	totalCost = l.totalCost
+	l.Unlock()
+	if del {
+		metrics.StatsCacheLRUCounter.WithLabelValues(typDel).Inc()
+		metrics.StatsCacheLRUMemUsage.WithLabelValues(typTrack).Set(float64(trackingCost))
+		metrics.StatsCacheLRUMemUsage.WithLabelValues(typTotal).Set(float64(totalCost))
+	}
+}
+
+func (l *internalLRUCache) del(key int64) bool {
+	element := l.elements[key]
+	if element == nil {
+		return false
+	}
+	delete(l.elements, key)
+	memUsage := element.Value.(*cacheItem).tblMemUsage
+	l.calculateCost(&statistics.TableMemoryUsage{}, memUsage)
+	l.maintainList(element, nil, &statistics.TableMemoryUsage{}, memUsage)
+	return true
+}
+
+// Cost returns the current cost
+func (l *internalLRUCache) Cost() int64 {
+	l.RLock()
+	defer l.RUnlock()
+	return l.totalCost
+}
+
+// Keys returns the current Keys
+func (l *internalLRUCache) Keys() []int64 {
+	l.RLock()
+	defer l.RUnlock()
+	r := make([]int64, 0, len(l.elements))
+	for _, v := range l.elements {
+		r = append(r, v.Value.(*cacheItem).key)
+	}
+	return r
+}
+
+// Values returns the current Values
+func (l *internalLRUCache) Values() []*statistics.Table {
+	l.RLock()
+	defer l.RUnlock()
+	r := make([]*statistics.Table, 0, len(l.elements))
+	for _, v := range l.elements {
+		r = append(r, v.Value.(*cacheItem).value)
+	}
+	return r
+}
+
+// Map returns the map of table statistics
+func (l *internalLRUCache) Map() map[int64]*statistics.Table {
+	l.RLock()
+	defer l.RUnlock()
+	r := make(map[int64]*statistics.Table, len(l.elements))
+	for k, v := range l.elements {
+		r[k] = v.Value.(*cacheItem).value
+	}
+	return r
+}
+
+// Len returns the current length
+func (l *internalLRUCache) Len() int {
+	l.RLock()
+	defer l.RUnlock()
+	return len(l.elements)
+}
+
+// FreshMemUsage re-calculate the memory message
+func (l *internalLRUCache) FreshMemUsage() {
+	var trackingCost int64
+	var totalCost int64
+	l.Lock()
+	for _, v := range l.elements {
+		item := v.Value.(*cacheItem)
+		oldMemUsage := item.tblMemUsage
+		newMemUsage := item.value.MemoryUsage()
+		item.tblMemUsage = newMemUsage
+		l.calculateCost(newMemUsage, oldMemUsage)
+		l.maintainList(v, nil, newMemUsage, oldMemUsage)
+	}
+	l.evictIfNeeded()
+	totalCost = l.totalCost
+	trackingCost = l.trackingCost
+	l.Unlock()
+	metrics.StatsCacheLRUMemUsage.WithLabelValues(typTrack).Set(float64(trackingCost))
+	metrics.StatsCacheLRUMemUsage.WithLabelValues(typTotal).Set(float64(totalCost))
+}
+
+// FreshTableCost re-calculate the memory message for the certain key
+func (l *internalLRUCache) FreshTableCost(key int64) {
```

Comment:
Where will the function be used? Is it like the following case:
We get one `statistics.Table` from cache and modify it. At that time we need to `FreshTableCost`. 
`FreshTableCost` seems play a similar role as `Put`.

Reply:
It will be used during initStats when we loaded cmsketch for each table each time.